modules/objc/examples/code_snippets/SampleCodeTest.m

```diff
@@ -1368,1143 +1376,1271 @@ - (void) dontTestListenerCertificateAuthenticatorCallback {
-  // Listener:
-  thisConfig.authenticator = ListenerCertificateAuthenticator.init (rootCerts: [rootCert])
+        CBLDocument* sameDoc = [db documentWithID: @"1000"];
+        CBLBlob* sameBlob = [sameDoc blobForKey: @"avatar"];
+        NSString* theBlobAsJSONstring = [sameBlob toJSON];
+        NSDictionary* dict = sameDoc.toDictionary;
+        for (id key in dict) {
+            NSLog(@"Data -- {%@) = {%@}", key, [dict valueForKey: key]);
+        }
 
-// end::old-listener-config-client-auth-root[]
-/
+        if ([CBLBlob isBlob: sameBlob.properties]) {
+            NSLog(@"%@", theBlobAsJSONstring);
+        }
 
-// tag::listener-config-client-auth-self-signed[]
-thisConfig.authenticator = ListenerCertificateAuthenticator.init {
-  (cert) -> Bool in
-    var cert:SecCertificate
-    var certCommonName:CFString?
-    let status=SecCertificateCopyCommonName(cert, &certCommonName)
-    if (self._allowlistedCommonNames.contains(["name": certCommonName! as String])) {
-        return true
-    }
-    return false
+    // end::tojson-blob[]
+    NSLog(@"print to avoid warning %@", theBlobAsJSONstringFails);
 }
 
-// end::listener-config-client-auth-self-signed[]
+- (void) dontTestToJSONDictionary {
+    // Example 6. Dictionaries as JSON strings
+    // tag::tojson-dictionary[]
+        NSString* aJSONstring = @"{\"id\":\"1002\",\"type\":\"hotel\",\"name\":\"Hotel Ned\","
+        "\"city\":\"Balmain\",\"country\":\"Australia\",\"description\":\"Undefined description for Hotel Ned\"}";
 
-// tag::p2p-ws-api-urlendpointlistener[]
-public class URLEndpointListener {
-    // Properties // <1>
-    public let config: URLEndpointListenerConfiguration
-    public let port UInt16?
-    public let tlsIdentity: TLSIdentity?
-    public let urls: Array<URL>?
-    public let status: ConnectionStatus?
-    // Constructors <2>
-    public init(config: URLEndpointListenerConfiguration)
-    // Methods <3>
-    public func start() throws
-    public func stop()
-}
+        NSError* error = nil;
+        CBLMutableDictionary* myDict = [[CBLMutableDictionary alloc] initWithJSON: aJSONstring error: &error];
+        NSLog(@"%@", myDict);
 
-// end::p2p-ws-api-urlendpointlistener[]
+        NSString* name = [myDict stringForKey: @"name"];
+        NSLog(@"Details for: %@", name);
 
+        for (NSString* key in myDict) {
+            NSLog(@"%@ %@", key, [myDict valueForKey: key]);
+        }
 
-// tag::p2p-ws-api-urlendpointlistener-constructor[]
-let config = URLEndpointListenerConfiguration.init(database: self.oDB)
-thisConfig.port = tls ? wssPort : wsPort
-thisConfig.disableTLS = !tls
-thisConfig.authenticator = auth
-self.listener = URLEndpointListener.init(config: config) // <1>
+    // end:tojson-dictionary[]
+}
 
-// end::p2p-ws-api-urlendpointlistener-constructor[]
+- (void) dontTestToJSONDocument {
+    // Example 7. Documents as JSON strings
+    // tag::tojson-document[]
+    NSError* error = nil;
```

Comment:
This example could be simplified as :

1. Get a document from a database.
2. Call toJSON
3. Print out or log JSON.

It is hard to understand when adding the complexity by query the doc id, get do, create doc and so on.